src/createStore.js

```diff
@@ -115,6 +116,8 @@ export default function createStore(reducer, initialState, enhancer) {
 
       isSubscribed = false
 
+      listener()
```

Comment:
This listener call is entirely unrelated to any state change. It can only be called outside of any dispatch (this is now enforced via #1569). It is impossible to have a state change where the listener is not called. 

This is basically creating exactly the thing this PR looks to avoid: an unnecessary listener call.